### ISSUE TITLE
Generalize content processing with markdown as default

### ISSUE THREAD

**Issue by jrehwaldt** [nikku/kartoffeldruck#4]
An option to add a post-processor would be nice. Basically, there could be a stack of content processors (per extension/glob pattern) with markdown being one of them, but that defeats the goal of an 'intentionally non-pluggable' static site generator. This would enable easy additions, which are not easily covered using nunjucks filters, such as emojis (#6).

---

_Update_: Made this issue focus on the generalization and extract the smiley request into #6.

**Comment by nikku** [comment]
Sounds good. Can yon create a PR on this?

**Comment by jrehwaldt** [comment]
Going good here :smile: Existing tests are green.

**Comment by jrehwaldt** [comment]
How do I merge my super duper change now... 23f98f75aef529a1842bb2dfa9bac9a5d737c574

**Comment by nikku** [comment]
An option (no guarantees): 

```
# gives you a-commit-hash
git commit ... 
git fetch upstream
git checkout master
git reset --hard upstream/master
# applies your change in a-commit-hash on new master (without history)
git cherry-pick a-commit-hash 
```

**Comment by jrehwaldt** [comment]
Warum einfach, wenn's auch kompliziert geht...

**Comment by nikku** [comment]
Git skillz, for da win, wa :wink:

**Comment by jrehwaldt** [comment]
I didn't see you mention rebase. Skill loss all the way :smile_cat:

**Comment by nikku** [comment]
:cake:

**Comment by jrehwaldt** [comment]
+1

**Comment by nikku** [comment]
PS: https://lefedt.de/blog/posts/2013/contributing-to-oss-through-pull-requests/

**Comment by jrehwaldt** [comment]
Didn't find it. https://lefedt.de/blog/posts/2016/how-to-handle-subsequent-and-dependent-pull-requests-if-the-oss-author-rewrites-them-when-merging/ ? ;-)

**Comment by nikku** [comment]
Pluggable content processors are beyond the scope of this project.

There are a lot alternative solutions out there that offer that functionality already, i.e [metalsmith](http://www.metalsmith.io/) that are 100% pluggable.

kartoffeldruck should work out of the box and do only one thing right: Flexible content generation with a powerful templating language in the back (Nunjucks) and a great way to write posts (Frontmatter + Markdown).

**Comment by nikku** [comment]
Smilies can still be a great feature, just because they make up a great addition for writing blog posts.

**Comment by jrehwaldt** [comment]
I can see this won't come so I'll close it...

**Comment by nikku** [comment]
Closed via f89d412.
